frontend/streamlit_app.py

```diff
@@ -146,62 +175,72 @@ def main() -> None:
         ):
             response, sources = response_tuple
             if response is not None:
-                response_buffer = ""
+                response_buffer = response
 
                 with st.chat_message("ai"):
                     message_placeholder = st.empty()
-
-                    response_buffer = ""
-                    for chunk in response.split(" "):
-                        response_buffer += chunk + " "
-                        if chunk.endswith("\n"):
-                            response_buffer += " "
-                        message_placeholder.markdown(response_buffer)
-                        time.sleep(0.05)
-
                     message_placeholder.markdown(response_buffer)
 
+                # Display response time
                 response_time_text = (
                     f"Response Time: {response_time / 1000:.2f} seconds"
                 )
-                response_time_colored = f":{'green' if response_time < 5000 else 'orange' if response_time < 10000 else 'red'}[{response_time_text}]"
-                st.markdown(response_time_colored)
+                if response_time < 5000:
+                    color = "green"
+                elif response_time < 10000:
+                    color = "orange"
+                else:
+                    color = "red"
+                st.markdown(
```

Comment:
This method is too unsafe. Consider this? https://discuss.streamlit.io/t/colored-text/34892/4

Reply:
Hi, I have updated to ` st.markdown(f":{color}[{response_time_text}]")
`